### ISSUE TITLE
Final Review Request of seven (7) W3C VCWG Specifications

### ISSUE THREAD

**Issue by msporny** [w3ctag/design-reviews#1029]
The W3C VCWG is requesting a *final* horizontal review of the following specifications, which your group has reviewed previous versions of over the past year or more.

We are requesting this before proceeding to the Proposed Recommendation phase based on a request by W3M to ensure that changes that have been made in the past six months are reviewed by your group since your last review. We are planning to enter the Proposed Recommendation phase for all seven (7) documents in Q1 2025 (roughly February, if possible).

 - [Verifiable Credentials Data Model v2.0](https://www.w3.org/TR/vc-data-model-2.0/)
   - [Changes since Candidate Recommendation](https://www.w3.org/TR/vc-data-model-2.0/#revision-history)
   - [Implementation Report](https://w3c.github.io/vc-data-model-2.0-test-suite/)
 - [Verifiable Credential Data Integrity v1.0](https://www.w3.org/TR/vc-data-integrity/)
   - [Changes since Candidate Recommendation](https://www.w3.org/TR/vc-data-integrity/#revision-history)
   - Implementation Report (see ECDSA and EdDSA test suites, which test the Data Integrity base spec in each cryptosuite)
 - [ECDSA Cryptosuite v1.0](https://www.w3.org/TR/vc-di-ecdsa/)
   - [Changes since Candidate Recommendation](https://www.w3.org/TR/vc-di-ecdsa/#revision-history)
   - [Implementation Report](https://w3c.github.io/vc-di-ecdsa-test-suite/)
 - [EdDSA Cryptosuite v1.0](https://www.w3.org/TR/vc-di-eddsa/)
   - [Changes since Candidate Recommendation](https://www.w3.org/TR/vc-di-eddsa/#revision-history)
   - [Implementation Report](https://w3c.github.io/vc-di-eddsa-test-suite/)
 - [Securing Verifiable Credentials using JOSE and COSE](https://www.w3.org/TR/vc-jose-cose/)
   - [Changes since Candidate Recommendation](https://github.com/w3c/vc-jose-cose/commits/?since=2024-09-15&until=now)
   - [Implementation Report](https://w3c.github.io/vc-jose-cose-test-suite/)
 - [Bitstring Status List v1.0](https://www.w3.org/TR/vc-bitstring-status-list/)
   - [Changes since Candidate Recommendation](https://www.w3.org/TR/vc-bitstring-status-list/#revision-history)
   - [Implementation Report](https://w3c.github.io/vc-bitstring-status-list-test-suite/)
 - [Controlled Identifier Documents v1.0](https://www.w3.org/TR/controller-document/)
   - [Changes since Candidate Recommendation](https://www.w3.org/TR/controller-document/#revision-history)
   - [Implementation Report](https://w3c.github.io/cid/implementations/1.0/)

  - Primary contacts:
      - Brent Zundel @brentzundel (Chair)
      - Ivan Herman @iherman (W3C Staff Contact)
      - Greg Bernstein @wind4greg (Editor, Data Integrity and cryptosuite specs)
      - Mike Jones @selfissued (Editor, VCDM v2.0 and VC JOSE COSE spec)
      - Gabe Cohen @decentralgabe (Editor, VC JOSE COSE spec)
      - Manu Sporny @msporny (Editor, VCDM v2.0, Data Integrity, Bitstring Status List specs)
      - Dmitri Zagidulin @dmitrizagidulin (Editor, Data Integrity and cryptosuite specs)
  - Organization/project driving the specification:
      - W3C Verifiable Credentials Working Group

Further details:

There are no major unresolved issues or opposition to these specifications that we know of at the present time.

**Comment by msporny** [comment]
> Could you please give us a rundown of any architectural changes that have been made in the past six months?

Yes, happy to -- the changes since the First Candidate Recommendation for each specification are directly linked to via the initial issue via the "Changes since Candidate Recommendation" link:

https://github.com/w3ctag/design-reviews/issues/1029#issue-2740828699

**Comment by marcoscaceres** [comment]
@msporny, any reason why the mime type is lacking +json?

**Comment by msporny** [comment]
@marcoscaceres wrote:
> @msporny, any reason why the mime type is lacking +json?

Grab your tissues, my friend, because this is going to be a really sad story with an unsatisfying end. :P

Yes, the topic of using a structured suffix came up 5+ years ago when some of us tried to register `application/did+ld+json`, which resulted in a: "That's legal, but we don't know what it means to have multiple plus signs." from the Designated Experts at IETF. That sent both the W3C DIDWG and VCWG on a side quest at IETF to determine if "+ld+json" was allowable as a suffix since the data model was JSON-LD and the media type for JSON-LD is `application/ld+json`... so, obviously, if a media type used JSON-LD as a format, the base media type would just add "+ld+json", right?

At the time, we thought we could just keep adding plus signs because RFC6838 allowed it syntactically... and we thought it was ok since, for JSON-LD, we were just asked to add `+json` to the base media type (`application/ld`) to state that the format was JSON... but then when we tried to repeat that pattern for `application/did` by doing `application/did+ld+json` the DEs were like: Hmm, no idea what that means. 

So, the IETF MEDIAMAN WG was chartered to answer that question and after years of discussion at W3C and IETF, we landed on this spec (which tried to explain the rules behind using multiple plus signs):

https://www.ietf.org/archive/id/draft-ietf-mediaman-suffixes-05.html

...and after more years of grueling debate at the IETF, the result was (consensus through exhaustion) to NOT allow the use of multiple structured suffixes, which resulted in this spec:

https://www.ietf.org/archive/id/draft-ietf-mediaman-suffixes-08.html

... which is being rolled into RFC6838bis as we speak. 

The side-effect of that decision meant that if we added "+json", we couldn't add any other sort of suffix in the future, and we already knew that people wanted to add other suffixes (like "+sd-jwt", or "+cborld"). So, the WG was like "We now hate suffixes because they've caused us nothing but pain over 5+ years AND because we know that we want to do +sd-jwt at some point and we're only allowed one plus sign now."... so we picked a media type that had no plus signs so this issue wouldn't bite future VCWGs or DIDWGs when they (inevitably) extended the media type.

Like I said, sad story with an unsatisfying ending. 

In order to salvage something positive from all of that, we are thinking of creating an off-broadway musical inspired by these events... working titles are currently "MIME and Punishment", "The Forbidden Plus", or "Cat+as+trophe". We'd bring the cast of Cats back for that last one, obviously.

**Comment by marcoscaceres** [comment]
Lol, ok. I want a role in this musical tho, or there will be a 🥁🎶 FoOOoormAAlll ObjeCCCttions 🥁. 

Ok, so seeing you've suffered enough, I don't have any further comments.

**Comment by msporny** [comment]
> I want a role in this musical tho

Done.

> 🥁🎶 FoOOoormAAlll ObjeCCCttion 🥁

Looks like we're gonna need the entire cast of Hamilton, too.

> so seeing you've suffered enough, I don't have any further comments.

Thank you, kind sir, this act of compassion will not soon be forgotten.

**Comment by marcoscaceres** [comment]
Closing as satisfied.

**Comment by marcoscaceres** [comment]
@msporny, what's the relationship of this issue to https://github.com/w3ctag/design-reviews/issues/899 ? (hehe, yes, do my job for me)

**Comment by msporny** [comment]
> [@msporny](https://github.com/msporny), what's the relationship of this issue to [#899](https://github.com/w3ctag/design-reviews/issues/899) ? 

#899 predates the request to review *Securing Verifiable Credentials using JOSE and COSE* in this issue. IOW, #899 is a duplicate of this issue and the [satisfaction](https://github.com/w3ctag/design-reviews/issues/1029#issuecomment-2735699135) of this review, IMHO, closes #899 as well. I see you came to the [same conclusion](https://github.com/w3ctag/design-reviews/issues/899#issuecomment-2735712608) in #899, so I think we're good here (no further action from TAG necessary).

> (hehe, yes, do my job for me)

I'll take it out of your positive karma balance... your remaining balance is around a trillion karma points. :)
